src/client.rs

```diff
@@ -55,12 +49,13 @@ impl SpiceClient {
     ///     let mut client = Client::new("API_KEY").await.unwrap();
     /// }
     /// ```
+    #[deprecated(note = "Use spiceai::ClientBuilder instead")]
```

Comment:
It's also idiomatic to add a `builder()` method on `SpiceClient` which returns a new builder. Then you can construct it like:

```rust
let client = SpiceClient::builder().build().await?;

Reply:
added, thanks